strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/configuration/ServerConfigurationController.java

```diff
@@ -253,6 +267,53 @@ public ResponseEntity setServerSettings(@RequestBody @Validated ServerSettingsFo
         return getSuccessfulResponseEntity(SUCCESSFUL_SAVE_SERVER_SETTINGS, acceptHeader);
     }
 
+    private void validateServerSettingsForm(ServerSettingsForm form,
+                                            BindingResult bindingResult)
+    {
+        if (form != null)
+        {
+            if (!isProxyConfigurationFormEmpty(form.getProxyConfigurationForm()))
+            {
+                ValidationUtils.invokeValidator(validator, form, bindingResult, ProxyConfigurationForm.ProxyConfigurationFormChecks.class);
+            }
+
+            if (!isSmtpConfigurationFormEmpty(form.getSmtpConfigurationForm()))
+            {
+                ValidationUtils.invokeValidator(validator, form, bindingResult, SmtpConfigurationForm.SmtpConfigurationFormChecks.class);
+            }
+
+            ValidationUtils.invokeValidator(validator, form, bindingResult);
+        }
+
+    }
+
+    private boolean isProxyConfigurationFormEmpty(ProxyConfigurationForm form)
+    {
+        return Stream.of(form.getHost(), form.getPort(), form.getType())
+                     .allMatch(this::isNullOrEmpty);
+    }
+
+    private boolean isSmtpConfigurationFormEmpty(SmtpConfigurationForm form)
+    {
+        return Stream.of(form.getHost(), form.getPort(), form.getConnection())
+                     .allMatch(this::isNullOrEmpty);
+    }
+
+    private boolean isNullOrEmpty(Object object)
```

Comment:
This doesn't work very well - try the following settings:
1. SMTP Settings
2. Encryption => anything other than `none` (make sure you've updated your `strongbox-web-ui` artifact, I've deployed a new version)
3. Change `SMTP Port` to nothing

Result is an error only for the `smtp host` and no error for `smtp port`. If you enter a `host`, you can save it with no `port`. The same applies for the `proxy` configuration as well. :)  

![image](https://user-images.githubusercontent.com/2303663/49767246-67413280-fce0-11e8-9376-c449fcf44b8f.png)

Reply:
Solved. There wasn't a validation annotation when port is empty.